File: RaftSwift/Sources/RaftCore/Utils/Types/ServerState.swift
/// The current role of the server in the Raft cluster
public enum ServerState: Sendable, Codable {
    /// Passive role; waits for messages from leader
    case follower

    /// Candidate trying to gather votes to become leader
    case candidate

    /// Active leader role that sends heartbeats and manages log replication
    case leader
}

File: RaftKotlin/raft-node/src/main/kotlin/raft/core/utils/types/ServerState.kt
package raft.core.utils.types

/**
 * The current role of the server in the Raft cluster
 */
enum class ServerState {
    /**
     * Passive role; waits for messages from leader
     */
    FOLLOWER,
    /**
     * Candidate trying to gather votes to become leader
     */
    CANDIDATE,
    /**
     * Active leader role that sends heartbeats and manages log replication
     */
    LEADER
}
